src/server/main_service.cc

```diff
@@ -2072,7 +2072,8 @@ GlobalState Service::GetGlobalState() const {
 
 void Service::ConfigureHttpHandlers(util::HttpListenerBase* base, bool is_privileged) {
   // We skip authentication on privileged listener if the flag admin_nopass is set
-  const bool should_skip_auth = is_privileged && !RequirePrivilegedAuth();
+  // We also skip authentication if requirepass is empty
+  const bool should_skip_auth = is_privileged && !RequirePrivilegedAuth() && !GetPassword().empty();
```

Comment:
in this change if is_privileged is false than should_skip_auth is false no matter if password was set

Reply:
I messed that up while refactoring + I was on a call at the same time :rofl: